cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/SystemArguments.java

```diff
@@ -336,6 +343,52 @@ public static Resources getResources(Map<String, String> args, @Nullable Resourc
     return builder;
   }
 
+  /**
+   * Adds any profile related arguments to the map
+   *
+   * @param args arguments to add to
+   * @param profile the profile to add
+   */
+  public static void addProfileArgs(Map<String, String> args, Profile profile) {
+    args.put(PROFILE_NAME, profile.getScopedName());
```

Comment:
I think it's better to have these two puts after the loop below, otherwise user can override these two.

Reply:
Also, what happen if a provisioner actually has a property called "system.profile.name"? Shall we store the profile name and profile provision in the system arguments instead of the user arguments?

Reply:
profile properties are prefixed with 'system.profile.properties.', so it would become 'system.profile.properties.system.profile.name', and the user would have to use that key in order to override it. I think having it in user arguments is consistent with other things like the resources and tx level, and having it there also indicates that it can be overridden by the user.

Reply:
hm though I guess we don't add those values to user arguments, we just allow them to be overridden from them. Though we do always add 'logical.start.time'. I guess I can move them to system arguments.